### ISSUE TITLE
Successful After hook results in scenario with skipped steps being marked as 'passed' instead of 'skipped'

### ISSUE THREAD

**Issue by dbarbuzzi** [cucumber/cucumber-js#1200]
With no hooks in place, a scenario with a skipped step is marked as ‘skipped’ in the overall run results. However, if that scenario has an `After` hook which completes successfully, the overall run results instead show that scenario as passed.

My personal expectation is that an `After` hook should either not be able to change the scenario status (i.e. used for teardown) or should only be able to change the status ‘negatively’ (e.g. change it in a negative direction such as from ‘passed’ to ‘failed’ but not in a positive direction such as from ‘skipped’ to ‘passed’). Ultimately, this misconception/belief may be the problem rather than any code/logic bug.

**Environment**

```
macOS 10.14.4
node 12.1.0
npm 6.9.0
cucumber.js 5.1.0
```

**To Reproduce**

A gist is linked below which takes the base example from the [documentation](https://github.com/cucumber/cucumber-js/blob/master/docs/nodejs_example.md) and makes a couple of minor modifications (both in `steps.js`):

1. A `return "skipped";` line is added to the `When` step
2. An `After` hook is added which simply does `return true;`

https://gist.github.com/dbarbuzzi/e161b91e27295c171496cb44ca1574a2

With those changes, the output of the test run is:

```sh
.--..--..--..--.

4 scenarios (4 passed)
12 steps (8 skipped, 4 passed)
0m00.000s
```

But I would have expected:

```sh
.--..--..--..--.

4 scenarios (4 skipped)
12 steps (8 skipped, 4 passed)
0m00.000s
```

**Comment by dbarbuzzi** [comment]
A quick follow-up: I have discovered a partially usable workaround by which any `After` hooks I can define/control can return the `scenario.result.status` for the `scenario` object they can receive as a parameter (should they otherwise complete successfully). However, I initially discovered this behavior when working with [protractor-cucumber-framework](https://github.com/protractor-cucumber-framework/protractor-cucumber-framework) because it is seemingly automatically injecting an `After` hook which I cannot modify.

**Comment by charlierudolph** [comment]
Thanks for the thorough write-up. Definitely a bug that should be fixed.

**Comment by mgonnet** [comment]
I think this was resolved in this PR: #1162 
Thanks @charlierudolph for merging!!
